rust/tools/authorization-logic/src/compilation_top_level.rs

```diff
@@ -46,11 +81,21 @@ pub fn source_file_to_ast_test_only(input_file_path: &str) -> AstProgram {
     source_file_to_ast(&resolved_in_file_path)
 }
 
-pub fn compile(input_file_path: &str, output_file_path: &str, decl_skip: &Vec<String>){
-    let resolved_in_file = utils::get_resolved_path(&input_file_path);
+pub fn compile_one_program(input_file_path: &str, output_file_path: &str,
+                           decl_skip: &Vec<String>) {
+    compile(&vec![input_file_path.to_string()], output_file_path,
+        decl_skip);
+}
+
+pub fn compile(input_file_paths: &Vec<String>, output_file_path: &str, decl_skip: &Vec<String>){
+    let resolved_in_files = input_file_paths.into_iter()
+        .map(|f| utils::get_resolved_path(f));
     let resolved_out_file = utils::get_resolved_path(&output_file_path);
-    let prog = source_file_to_ast(&resolved_in_file);
-    let prog_with_imports = import_assertions::handle_imports(&prog);
+    let progs = resolved_in_files
+        .map(|f| source_file_to_ast(&f))
+        .collect();
+    let merged_prog = merge_multiprogram(progs);
+    let prog_with_imports = import_assertions::handle_imports(&merged_prog);
```

Comment:
This might factor the complexity better to consider compiling a multiprogram to be a composition of combining the programs and compiling that one program. That would leave `compile` largely unchanged, but would add a `compile_multiprogram` function as an optional previous step.

Reply:
I think this suggestion is about the same as what's here except the names of functions, and the code that previously made use of `compile` has already been changed to use `compile_one_program`, so I'd rather leave it as-is.